### PR TITLE
fix(account-lib): fix types and overrestricted types

### PR DESCRIPTION
**Pull request by charly-simtlix**
Issues: STLX-9929

Buffer were used in several places were an Uint8Array is expected. ALGO were using Uint8Array so when migrated to TS type errors were found.

Also some typos in comments were fixed.

One function was refactored a little to avoid duplicating the decoding.

### CODE REVIEW

File: modules/account-lib/src/coin/algo/utils.ts [hunk 1]
```diff
@@ -294,8 +295,26 @@ export class Utils implements BaseUtils {
...
+  tryToDecodeUnsignedTransaction(buffer: Buffer): EncodedTx | null {
+    try {
+      const txn = algosdk.decodeUnsignedTransaction(buffer);
+      return {
+        rawTransaction: new Uint8Array(buffer),
+        txn,
+        signed: false,
+      };
+    } catch (_: unknown) {
+      return null;
```
Comment (JulioAchaval-SimTLiX) [inline modules/account-lib/src/coin/algo/utils.ts:317]: shouldn't this catch throw an exception?

**Review by JulioAchaval-SimTLiX — APPROVED** [review]

File: modules/account-lib/src/coin/algo/utils.ts [hunk 2]
```diff
@@ -159,9 +159,9 @@ export class Utils implements BaseUtils {
   /**
    * decodeSeed decodes an algo seed
    *
-   * Decoding algo seed is sane as decoding address.
-   * Latest version of algo sdk (1.9, at this writing) does not expose explicit method for decoding seed
-   * hence, this routine uses decodeAddress and changes the return structure
+   * Decoding algo seed is same as decoding address.
+   * Latest version of algo sdk (1.9, at this writing) does not expose explicit method for decoding seed.
+   * Parameters is base 32 decoded and split into seed and checksum.
```
Comment (tylerlevine) [inline modules/account-lib/src/coin/algo/utils.ts:164]:
```suggestion
   * Parameter is either hex or base64 decoded and split into seed and checksum.
```

File: modules/account-lib/src/coin/algo/utils.ts [hunk 3]
```diff
@@ -261,44 +255,69 @@ export class Utils implements BaseUtils {
...
-          const addr = algosdk.encodeAddress(sig.pk);
-          signers.push(addr);
-          if (sig.s) {
-            signedBy.push(addr);
-          }
-        }
-      }
+    try {
+      return this.tryToDecodeUnsignedTransaction(buffer);
+    } catch (_: unknown) {
```
Comment (tylerlevine) [inline modules/account-lib/src/coin/algo/utils.ts:260]: https://mariusschulz.com/blog/optional-catch-binding-in-typescript

```suggestion
    } catch {
```

File: modules/account-lib/src/coin/algo/utils.ts [hunk 4]
```diff
@@ -261,44 +255,69 @@ export class Utils implements BaseUtils {
...
-        rawTransaction: new Uint8Array(buffer),
-        txn: tx.txn,
-        signed: true,
-        signers: signers,
-        signedBy: signedBy,
-      };
-    } else {
+    try {
+      return this.tryToDecodeSignedTransaction(buffer);
+    } catch (_: unknown) {
```
Comment (tylerlevine) [inline modules/account-lib/src/coin/algo/utils.ts:266]:
```suggestion
    } catch {
```

**Review by tylerlevine — APPROVED** [review]
good clean up, very minor nits

**Comment by shihao-guo** [comment]
do we need unit tests?

**Comment by charly-simtlix** [comment]
>

**Review by tylerlevine — APPROVED** [review]

**Pull request merged**